core/trino-main/src/main/java/io/trino/operator/exchange/LocalExchange.java

```diff
@@ -67,6 +68,10 @@
 public class LocalExchange
 {
     private static final int SCALE_WRITERS_MAX_PARTITIONS_PER_WRITER = 128;
+    // This represents the percentage of maxMemoryPerNode upon which we will stop scaling of writers locally. We have
```

Comment:
nit: I don't think you need `SCALE_WRITERS_MAX_MEMORY_RATIO` since it's not exposed to users. Just set 256MB